java/ql/src/experimental/semmle/code/xml/StrutsXML.qll

```diff
@@ -0,0 +1,40 @@
+import java
+
+/**
+ * A deployment descriptor file, typically called `structs.xml`.
+ */
+class StrutsXMLFile extends XMLFile {
+  StrutsXMLFile() {
+    count(XMLElement e | e = this.getAChild()) = 1 and
+    this.getAChild().getName() = "struts"
+  }
+}
+
+/**
+ * An XML element in a `StrutsXMLFile`.
+ */
+class StrutsXMLElement extends XMLElement {
+  StrutsXMLElement() { this.getFile() instanceof StrutsXMLFile }
+
+  /**
+   * Gets the value for this element, with leading and trailing whitespace trimmed.
+   */
+  string getValue() { result = allCharactersString().trim() }
+}
+
+/**
+ * A `<constant>` element in a `struts.xml` file.
```

Comment:
```suggestion
 * A `<constant>` element in a `StrutsXMLFile`.
```
Might be better to refer to the QL class here, especially since that class says the file may not be named `struts.xml`.